### PR TITLE
ENH: Add handler for AreaDetector data.

### PR DESCRIPTION
**Pull request by danielballan**
This adds a handler to read area detector data in the format spec implemented by @stuwilkins and dubbed 'AD_HDF5'.

The reader is called `HDF5DatasetSliceHandler`, meaning to suggest that it slices into a single Dataset in an HDF5 file. The dataset name, `'/entry/data/data/'`, is hard-coded.

@stuwilkins Instead of hard-coding the dataset name, could your insertion code pass it to `__init__`? Also, it seems that `frame_per_point` is not used. Should that be removed from the `__init__`.

### CODE REVIEW

File: filestore/file_readers.py [hunk 1]
```diff
@@ -11,7 +11,33 @@
...
+    def close(self):
+        super(HDF5HandlerBase, self).close()
+        self._file.close()
+
+
+class HDF5DatasetSliceHandler(_HDF5HandlerBase):
+    "Handler for Stuart's first detector demo"
+    def __init__(self, filename, frame_per_point):
+        self._filename = filename
+        self.open()
```
Comment (tacaswell) [inline filestore/file_readers.py]: I would add `self._dataset = None` here

File: filestore/file_readers.py [hunk 2]
```diff
@@ -11,7 +11,33 @@
...
+
+    def __call__(self, point_number):
+        dataset_name = '/entry/data/data'
+        # Don't read out the dataset until it is requested for the first time.
+        if not hasattr(self, '_dataset'):
```
Comment (tacaswell) [inline filestore/file_readers.py]: and make this `if not self._dataset`

File: filestore/file_readers.py [hunk 3]
```diff
@@ -11,7 +11,33 @@
...
+            self._dataset = self._file[dataset_name]
+        return self._dataset[point_number, :, :]
```
Comment (tacaswell) [inline filestore/file_readers.py]: This should be `[(point_number * self._fpp): (point_number+1)*self._fpp, :, :].squeeze()`

File: filestore/file_readers.py (hunk 1 above)
Comment (tacaswell) [inline filestore/file_readers.py]: and you need to stash `frame_per_point` and default to `1`

Comment (danielballan) [inline filestore/file_readers.py]: Ah, I had not understood the intention of `frame_per_point`. Got it.

**Comment by danielballan** [comment]
This PR now includes:

* giving handlers a `specs` attribute with the set of the names of the spec(s) each handler is designed to read
* a test on data formatted to Stuart's 'AD_HDF5' spec

It also incorporates @tacaswell's bugfix in #28 into `db_disconnect` so that it does not have to be incorporated into each test separately.

File: filestore/file_readers.py [hunk 4]
```diff
@@ -11,7 +11,71 @@
...
+        self._fpp = frame_per_point
+        self._filename = filename
+        self._key = key
+        self._file = None
+        self._dataset = None
+        self.open()
+
+    def __call__(self, point_number):
+        # Don't read out the dataset until it is requested for the first time.
+        if not self._dataset:
```
Comment (tacaswell) [inline filestore/file_readers.py]: To be picky. this does not actually read the dataset into memory, it just sets up the objects to make sure that you can access the data.

File: filestore/file_readers.py [hunk 5]
```diff
@@ -11,7 +11,71 @@
...
+            self._dataset = self._file[self._key]
+        start, stop = point_number * self._fpp, (point_number + 1) * self._fpp
+        return self._dataset[start:stop].squeeze()
```
Comment (tacaswell) [inline filestore/file_readers.py]: This is the step where you actually read anything from disk.

**Pull request merged**